### DISCUSSION TITLE
Wireguard mode + Upstream proxy

### DISCUSSION THREAD

**Discussion by itispip** [mitmproxy/mitmproxy#5751] [Q&A]
Can wireguard mode work with upstream mode?

Currently if using --mode wireguard --mode upstream http://xxx:port, it actually provides both the 2 mode separately, while the expected behaviour is client -->  wireguard --> 2nd proxy --> internet

Environment is Win10.

**Comment by mhils** [comment]
> Can wireguard mode work with upstream mode?

Currently no. If you have HTTP only, you can probably rewrite `server_conn.via` with an addon (although I haven't tested that).

**Comment by IgorDuino** [comment]
Will be so usefull
